core/src/trie_pos.rs

```diff
@@ -144,6 +148,29 @@ impl TriePosition {
         bit
     }
 
+    /// Get the page ID this position lands in. Returns `None` at the root.
+    pub fn page_id(&self) -> Option<PageId> {
+        if self.is_root() {
+            return None;
+        }
+
+        let mut page_id = ROOT_PAGE_ID;
+        for (i, chunk) in self.path().chunks(DEPTH).enumerate() {
+            if i * DEPTH == self.depth as usize {
```

Comment:
The algorithm is this:
  1. take the next sextet.
  2. if there are no more bits left, return the current page
  3. if there are more bits left, push the page ID given by that sextet
  4. go to (1)

> with i * DEPTH == self.depth as usize it early return only if TriePosition is the last node of a page,

Yes, this is what we want, though, isn't it? If the TriePosition is at the last level of the page we don't want to push the next page ID, because it's not in that page.

> otherwise it would reach the page_id of the last page in the page_tree

We only take as many pages as there are sextets in the trie position

Reply:
Oh... I spent quite some time without getting it... because I was reading `self.path.chunks_exact...` instead of `self.path().chunks_exact...` and the `path` method was hiding the part where the path is just evaluated up to depth